src/squaregrid.jl

```diff
@@ -0,0 +1,57 @@
+export SquareGrid
+
+"""
+    SquareGrid(; kwargs...)(adj_matrix)
+    layout(algo::SquareGrid, adj_matrix)
+
+Position nodes on a 2 dimensional rectagular grid. The nodes are palced in order
+from upper left to lower right. To skip positions see `skip` argument.
+
+Takes adjacency matrix representation of a network and returns coordinates of
+the nodes.
+
+## Keyword Arguments
+- `Ptype=Float64`: Determines the output type `Point{2,Ptype}`
+- `cols=:auto`: Columns of the grid, the rows are determined automatic. If `:auto` the layout will be square-ish.
+- `dx=Ptype(1), dy=Ptype(-1)`: Ofsets between rows/cols.
+- `skip=Tuple{Int,Int}[]`: Specify positions to skip when placing nodes.
+  `skip=[(i,j)]` means to keep the position in the `i`-th row and `j`-th column
+  empty.
+"""
+struct SquareGrid{Ptype,CT} <: AbstractLayout{2,Ptype}
```

Comment:
To be honest, I am a bit unsure what the purpose of this layout is, does it really make sense to line up the vertices row by row on a grid? Won't this cause a huge mess when plotting the graph with overlapping edges?

Reply:
Well this one is quite specific for my usecase to be honest. In papers about networks with geographic/planar embedding (for example power grids) it is quite common to use square grids as toy models. The example in the docs is bad though.

<img width="300" alt="grafik" src="https://user-images.githubusercontent.com/35867212/121037890-2098fa00-c7b0-11eb-931b-af5f97786fe5.png">

(Franz Kaiser et al 2020 New J. Phys. 22 013053)

<img width="300" alt="grafik" src="https://user-images.githubusercontent.com/35867212/121037753-0101d180-c7b0-11eb-9fd9-98ccfccd9d35.png">

(Manik, D., Molkenthin, N. Topology dependence of on-demand ride-sharing. Appl Netw Sci 5, 49 (2020). https://doi.org/10.1007/s41109-020-00290-2)